app/controllers/nonsipp/landorproperty/LandPropertyIndependentValuationController.scala

```diff
@@ -0,0 +1,83 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.nonsipp.landorproperty
+
+import controllers.actions._
+import controllers.nonsipp.landorproperty.LandPropertyIndependentValuationController._
+import forms.YesNoPageFormProvider
+import models.Mode
+import models.SchemeId.Srn
+import navigation.Navigator
+import pages.nonsipp.landorproperty.LandPropertyIndependentValuationPage
+import play.api.data.Form
+import play.api.i18n.{I18nSupport, MessagesApi}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import services.SaveService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
+import viewmodels.DisplayMessage.Message
+import viewmodels.implicits._
+import viewmodels.models.{FormPageViewModel, YesNoPageViewModel}
+import views.html.YesNoPageView
+
+import javax.inject.{Inject, Named}
+import scala.concurrent.{ExecutionContext, Future}
+
+class LandPropertyIndependentValuationController @Inject()(
+  override val messagesApi: MessagesApi,
+  saveService: SaveService,
+  @Named("non-sipp") navigator: Navigator,
+  identifyAndRequireData: IdentifyAndRequireData,
+  formProvider: YesNoPageFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: YesNoPageView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  private val form = LandPropertyIndependentValuationController.form(formProvider)
+
+  def onPageLoad(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn) { implicit request =>
+    val preparedForm = request.userAnswers.fillForm(LandPropertyIndependentValuationPage(srn), form)
+    Ok(view(preparedForm, viewModel(srn, mode)))
+  }
+
+  def onSubmit(srn: Srn, mode: Mode): Action[AnyContent] = identifyAndRequireData(srn).async { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => Future.successful(BadRequest(view(formWithErrors, viewModel(srn, mode)))),
+        value =>
+          for {
+            updatedAnswers <- Future.fromTry(request.userAnswers.set(LandPropertyIndependentValuationPage(srn), value))
+            _ <- saveService.save(updatedAnswers)
+          } yield Redirect(navigator.nextPage(LandPropertyIndependentValuationPage(srn), mode, updatedAnswers))
+      )
+  }
+}
+
+object LandPropertyIndependentValuationController {
+  def form(formProvider: YesNoPageFormProvider): Form[Boolean] = formProvider(
+    "landPropertyIndependentValuation.error.required"
+  )
+
+  def viewModel(srn: Srn, mode: Mode): FormPageViewModel[YesNoPageViewModel] =
+    YesNoPageViewModel(
+      "landPropertyIndependentValuation.title",
+      Message("landPropertyIndependentValuation.heading", "1 Street Road"),
```

Comment:
As discussed @narendrapaduchuri please check if this now can come from the user answers.